modules/git/repo_commit.go

```diff
@@ -5,13 +5,19 @@
 package git
 
 import (
+	"bufio"
 	"bytes"
+	"context"
+	"fmt"
 	"io"
+	"os"
 	"strconv"
 	"strings"
 
 	"code.gitea.io/gitea/modules/cache"
+	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
```

Comment:
I think this is the first git module depend on structs. Maybe it's not a good idea. We can move function `ExtendedCommitStats` to service layer.

Reply:
From docs:

> modules: Different modules to handle specific functionality in Gitea. Work in Progress: Some of them should be moved to services, in particular those that depend on models because they rely on the database.

I don't have any dependency to models. Why it is not a good idea?

Reply:
Fixed in [51a9e8d](https://github.com/go-gitea/gitea/pull/27882/commits/51a9e8d5ecb0c6c9e1455794b37a71dec6a3ef07). But I still don't think new version is any better.